test/OpenTelemetry.Extensions.Docker.Tests/Resources/DockerResourceDetectorTests.cs

```diff
@@ -24,88 +25,168 @@ namespace OpenTelemetry.Extensions.Docker.Tests;
+            ExpectedContainerId = "dc64b5743252dbaef6e30521c34d6bbd1620c8ce65bdb7bf9e7143b61bb5b183",
+            CgroupVersion = DockerResourceDetector.ParseMode.V2,
+        },
+        new TestCase()
+        {
+            Name = "cgroupv2 with minikube containerd mountinfo",
+            Line = "1537 1517 8:1 /var/lib/containerd/io.containerd.grpc.v1.cri/sandboxes/fb5916a02feca96bdeecd8e062df9e5e51d6617c8214b5e1f3ff9320f4402ae6/hostname /etc/hostname rw,relatime - ext4 /dev/sda1 rw",
+            ExpectedContainerId = "fb5916a02feca96bdeecd8e062df9e5e51d6617c8214b5e1f3ff9320f4402ae6",
+            CgroupVersion = DockerResourceDetector.ParseMode.V2,
+        },
+        new TestCase()
+        {
+            Name = "cgroupv2 with minikube docker mountinfo",
+            Line = "2327 2307 8:1 /var/lib/docker/containers/a1551a1d7e1881d6c18d2c9ec462cab6ad3666825f0adb2098e9d5b198fd7e19/hostname /etc/hostname rw,relatime - ext4 /dev/sda1 rw",
+            ExpectedContainerId = "a1551a1d7e1881d6c18d2c9ec462cab6ad3666825f0adb2098e9d5b198fd7e19",
+            CgroupVersion = DockerResourceDetector.ParseMode.V2,
+        },
+        new TestCase()
+        {
+            Name = "cgroupv2 with minikube docker mountinfo2",
+            Line = "929 920 254:1 /docker/volumes/minikube/_data/lib/docker/containers/0eaa6718003210b6520f7e82d14b4c8d4743057a958a503626240f8d1900bc33/hostname /etc/hostname rw,relatime - ext4 /dev/vda1 rw",
+            ExpectedContainerId = "0eaa6718003210b6520f7e82d14b4c8d4743057a958a503626240f8d1900bc33",
+            CgroupVersion = DockerResourceDetector.ParseMode.V2,
+        },
+        new TestCase()
+        {
+            Name = "cgroupv2 with podman mountinfo",
+            Line = "1096 1088 0:104 /containers/overlay-containers/1a2de27e7157106568f7e081e42a8c14858c02bd9df30d6e352b298178b46809/userdata/hostname /etc/hostname rw,nosuid,nodev,relatime - tmpfs tmpfs rw,size=813800k,nr_inodes=203450,mode=700,uid=1000,gid=1000",
+            ExpectedContainerId = "1a2de27e7157106568f7e081e42a8c14858c02bd9df30d6e352b298178b46809",
+            CgroupVersion = DockerResourceDetector.ParseMode.V2,
+        },
+    };
+
+    private readonly List<TestCase> testInvalidCases = new()
+    {
+        new TestCase()
+        {
+            Name = "Invalid cgroupv1 line",
+            Line = "13:name=systemd:/podruntime/docker/kubepods/ac679f8a8319c8cf7d38e1adf263bc08d23zzzz",
+            CgroupVersion = DockerResourceDetector.ParseMode.V1,
+        },
+        new TestCase()
+        {
+            Name = "Invalid hex cgroupv2 line (contains a z)",
+            Line = "13:name=systemd:/var/lib/containerd/io.containerd.grpc.v1.cri/sandboxes/fb5916a02feca96bdeecd8e062df9e5e51d6617c8214b5e1f3fz9320f4402ae6/hostname",
+            CgroupVersion = DockerResourceDetector.ParseMode.V1,
```

Comment:
Isn't it test for `DockerResourceDetector.ParseMode.V2`

Reply:
Yes, let me fix that.